client/components/MessageList/index.js

```diff
@@ -1,168 +1,165 @@
-        },
-        {
-          id: 6,
-          author: 'apple',
-          message: 'It looks like it wraps exactly as it is supposed to. Lets see what a reply looks like!',
-          timestamp: new Date().getTime()
-        },
-        {
-          id: 7,
-          author: 'orange',
-          message: 'Hello world! This is a long message that will hopefully get wrapped by our message bubble component! We will see how well it works.',
-          timestamp: new Date().getTime()
-        },
-        {
-          id: 8,
-          author: 'orange',
-          message: 'It looks like it wraps exactly as it is supposed to. Lets see what a reply looks like!',
-          timestamp: new Date().getTime()
-        },
-        {
-          id: 9,
-          author: 'apple',
-          message: 'Hello world! This is a long message that will hopefully get wrapped by our message bubble component! We will see how well it works.',
-          timestamp: new Date().getTime()
-        },
-        {
-          id: 10,
-          author: 'orange',
-          message: 'It looks like it wraps exactly as it is supposed to. Lets see what a reply looks like!',
-          timestamp: new Date().getTime()
-        },
-      ]
-      setMessages([...messages, ...tempMessages])
-  }
-
-  const renderMessages = () => {
-    let i = 0;
-    let messageCount = messages.length;
-    let tempMessages = [];
-
-    while (i < messageCount) {
-      let previous = messages[i - 1];
-      let current = messages[i];
-      let next = messages[i + 1];
-      let isMine = current.author === MY_USER_ID;
-      let currentMoment = moment(current.timestamp);
-      let prevBySameAuthor = false;
-      let nextBySameAuthor = false;
-      let startsSequence = true;
-      let endsSequence = true;
-      let showTimestamp = true;
-
-      if (previous) {
-        let previousMoment = moment(previous.timestamp);
-        let previousDuration = moment.duration(currentMoment.diff(previousMoment));
-        prevBySameAuthor = previous.author === current.author;
-        
-        if (prevBySameAuthor && previousDuration.as('hours') < 1) {
-          startsSequence = false;
-        }
-
-        if (previousDuration.as('hours') < 1) {
-          showTimestamp = false;
-        }
+export default function MessageList({ profile }) {
+
+    const toRender = useRef('');
+    const messageState = useSelector(s => s.messages);
+    const [message, setMessage] = useState('');
+    const dispatch = useDispatch();
+    
+    useEffect(() => {
+      toRender.current = renderMessages(messageState);
+
+      if (messageState.isSuccessful) {
+        let echo = new Echo({
+          broadcaster: 'pusher',
+          key: 'f607f988d506ff262620',
+          cluster: 'eu',
+          forceTLS: false,
+          authorizer: (channel, options) => {
+            return {
+              authorize: (socketId, callback) => {
+                axios.post('http://127.0.0.1:8000/api/broadcasting/auth', {
```

Comment:
we have "localhost" here. I think you might want to implement this with redux and use the production URL instead of localhost. I also think it can use some refactoring, you could an util websocket file that handles these and import to for use in this component

Reply:
Refactored